### ISSUE TITLE
Little override issues and documentation

### ISSUE THREAD

**Issue by M0ebiu5** [bartbutenaers/node-red-contrib-msg-resend#6]
The standard nodes only allow an override via msg property, if the node property is not set.
But this resend node does not accept an empty interval. It's not documented, what has to be entered into the node properties, when using the override mechanism.

Also, there is no docu for the override in the side-panel (and no link to the flows documentation).

And i could not find the info, how to stop the resending process (for one or all topics).  A "reset" ("stop") and "reset all" command would maybe make sense.

Thanks for that nice node, which fills a gap in the standard nodes (trigger interval is not overridable)!

**Comment by bartbutenaers** [comment]
Hi @M0ebiu5,

### Override resend interval
Well this was my first node I have ever developed for Node-RED, so at the time being I was not aware that a message could only override **empty** node properties.  I have taken that into account for my recent nodes, but cannot change it for this one (to avoid existing flows running into troubles).  So you can set a new interval in the input message, even when it is specified in the node's config screen:
![image](https://user-images.githubusercontent.com/14224149/46171662-739d1e00-c2a1-11e8-8ca5-e7b356558840.png)
```
[{"id":"6da598bc.49b488","type":"inject","z":"279b8956.27dfe6","name":"Inject value to resend","topic":"","payload":"someValue","payloadType":"str","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":2460,"y":180,"wires":[["aff89c1c.5a51b"]]},{"id":"aff89c1c.5a51b","type":"msg-resend","z":"279b8956.27dfe6","interval":"3","maximum":5,"bytopic":false,"clone":false,"name":"","x":2850,"y":180,"wires":[["9d04065d.bf1c18"]]},{"id":"9d04065d.bf1c18","type":"debug","z":"279b8956.27dfe6","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","x":3030,"y":180,"wires":[]},{"id":"127f5d22.7b8f43","type":"inject","z":"279b8956.27dfe6","name":"Set interval to 2","topic":"","payload":"2","payloadType":"num","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":2440,"y":100,"wires":[["2d8e589b.b106f8"]]},{"id":"2d8e589b.b106f8","type":"change","z":"279b8956.27dfe6","name":"","rules":[{"t":"set","p":"resend_interval","pt":"msg","to":"payload","tot":"msg"},{"t":"set","p":"resend_ignore","pt":"msg","to":"true","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":2660,"y":140,"wires":[["aff89c1c.5a51b"]]},{"id":"69328ade.ede5d4","type":"inject","z":"279b8956.27dfe6","name":"Set interval to 4","topic":"","payload":"4","payloadType":"num","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":2440,"y":140,"wires":[["2d8e589b.b106f8"]]}]
```
**TODO:** I will need to add this info to the documentation.

### Link to the flows documentation
Do you mean that I should a link to my Github readme file, inside the side-panel ?

### Stop the resending process
I have added now a [section](https://github.com/bartbutenaers/node-red-contrib-msg-resend#stop-resending) about it in the readme page, but seems I indeed am **not** able to stop **all topics** currently (so the explanation in the readme is not entirely correct ...).

The current stop-resending mechanism works like this:
1.  When a new message arrives, the previous timer will be stopped. 
    A. Topic independent: there is only one timer running, so that will be stopped.
    B. Topic independent and `msg.topic` is filled: only the timer for that topic will be stopped.
    C. Topic independent and  `msg.topic` is empty: no timers will be stopped (since the documentation says: *'in case of topic dependent resending, messages without topic will be ignored'*).  
2. Then a new timer will be started, except when `msg.resend_ignore = true`: I introducted this property to allow users to override node properties by input messages, without resending those input messages.

So by adding `msg.resend_ignore = true` to your input message, the current timer will be stopped and NO new timer will be started.  That is the current way to stop resending for cases A & B but not for case C.  Your proposal with `msg.reset` makes sense, but then I have two **different ways** of working (which I would like to avoid).  But currently the node has no way to execute your `msg.reset_all`.  Will have to think about that...

Kind regards,
Bart

**Comment by M0ebiu5** [comment]
About the override: thanks for clarifying. From my point of view, the override of non empty fields also would be better - we lost such a nice way to provide default values...

About the link: i would like to have the link to github and/or [https://flows.nodered.org/node/node-red-contrib-msg-resend](url) (they are not identical?) in the side panel, cause usually there is more information about the node. If i want to reach it now, i have to go over the palette manager...

Thanks for answering my questions!

**Comment by ryancasler** [comment]
<strike>So, does this node not work then?   I can't seem to get the resend to ever stop.  It just restarts again.  If it doesn't work, you should pull it down so that folks don't go crazy thinking it works.</strike>
Nevermind.  I was setting the message to msg.msg.resend_ignore.   DOH!!!!!  #facepalm

**Comment by bartbutenaers** [comment]
Hi Ryan,
No problem.  Apologies accepted ;-)
Bart

**Comment by bartbutenaers** [comment]
HI @M0ebiu5,
I'm going to close this issue.  Last week I have published a new version (1.0.0) on npm, which has some new msg control options (see [readme](https://github.com/bartbutenaers/node-red-contrib-msg-resend#node-control) in case you might ever need it).
Kind regards,
Bart
